pytensor/link/pytorch/dispatch/random.py

```diff
@@ -0,0 +1,81 @@
+from copy import deepcopy
+from functools import singledispatch
+
+import numpy as np
+import torch
+from numpy.random import Generator
+
+import pytensor.tensor.random.basic as ptr
+from pytensor.link.pytorch.dispatch.basic import pytorch_funcify, pytorch_typify
+
+
+@pytorch_typify.register(Generator)
+def pytorch_typify_Generator(rng, **kwargs):
+    # XXX: Check if there is a better way.
+    # Numpy uses PCG64 while Torch uses Mersenne-Twister (https://github.com/pytorch/pytorch/blob/main/aten/src/ATen/CPUGeneratorImpl.cpp)
+    state = rng.__getstate__()
+    rng_copy = np.random.default_rng()
+    rng_copy.bit_generator.state = rng.bit_generator.state
+    seed = torch.from_numpy(rng_copy.integers([2**32]))
+    state["pytorch_gen"] = torch.manual_seed(seed)
```

Comment:
I don't think we need this monkeypatching on dictionaries anymore, just work directly with the `torch.manual_seed`

Reply:
i guess at this point in the pipeline, it doesn't matter if it is still a torch generator, since we're calling typify